internal/deb/extract_test.go

```diff
@@ -352,6 +352,62 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Dangling hard link",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Hln(0644, "./link", "./non-existing-target"),
+	}),
+	options: deb.ExtractOptions{
+		Extract: map[string][]deb.ExtractInfo{
+			"/**": []deb.ExtractInfo{{
+				Path: "/**",
+			}},
+		},
+	},
+	error: `cannot extract from package "test-package": link target does not exist: \/[^ ]*\/non-existing-target`,
+}, {
+	summary: "Hard link to symlink does not follow symlink",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Lnk(0644, "./symlink-to-file", "./file"),
+		testutil.Hln(0644, "./link-to-symlink", "./symlink-to-file"),
```

Comment:
```suggestion
		testutil.Lnk(0644, "./symlink", "./file"),
		testutil.Hln(0644, "./hardlink-to-symlink", "./symlink"),
```

Reply:
Fixed in [03ce4d3](https://github.com/letFunny/chisel/pull/17/commits/03ce4d322ccaff4f6ba8c00a32aee039de12c9d3).